libraries/AP_Camera/AP_Camera.cpp

```diff
@@ -251,17 +304,50 @@ MAV_RESULT AP_Camera::handle_command_long(const mavlink_command_long_t &packet)
         }
         return MAV_RESULT_DENIED;
     case MAV_CMD_IMAGE_START_CAPTURE:
-        if (!is_zero(packet.param2) || !is_equal(packet.param3, 1.0f) || !is_zero(packet.param4)) {
-            // Its a multiple picture request
-            if (is_equal(packet.param3, 0.0f)) {
+        // check if this is a single picture request
+        if (is_equal(packet.param3, 1.0f)) {
+            if (is_zero(packet.param1)) {
+                // take pictures for every backend
+                take_picture();
+                return MAV_RESULT_ACCEPTED;
+            }
+            if (take_picture(packet.param1-1)) {
+                return MAV_RESULT_ACCEPTED;
+            }
+        }
+        // multiple picture request
+        else if (is_zero(packet.param3)) {
```

Comment:
Little formatting issue here, the else should be placed on the same line as the previous curly bracket.  I know it makes placement of the comment more difficult but..

Reply:
done